### DISCUSSION TITLE
Confuse about the way import module on Deno project

### DISCUSSION THREAD

**Discussion by hh54188** [denoland/deno#23900] [Q&A]
As far as I know there are 4 ways of import module in Deno project:

- deno.land/x
- esm.sh
- jsr
- npm

And I guess the priority order would be just like how I list them above? I mean deno official recommend using deno.land rather than esm.sh, esm.sh would be a better choice than jsr.

Suppose I was right, here is the confusing part when I try to install Hono and its CORS middleware.

- I can find Hono framework both on NPM and deno.land, but I cannot find ExpressJS on deno.land by searching [on this page](https://deno.land/x). What’s the standard of deno.land including the third party modules? I assume ExpressJS is popular among the NodeJS community.
- Suppose esm.sh is the second best way to import modules because they should include all standard ES modules. However, I cannot find the appropriate way to import Hono CORS middleware via esm.sh.
- 
This is the way show on [official site](https://hono.dev/middleware/builtin/cors) about how to introduce on deno.land

```javascript
import { cors } from 'https://deno.land/x/hono/middleware.ts'. 
```

But I cannot find the module on esm.sh:

```javascript
https://esm.sh/hono@4.3.8/middleware 
```
- What’s purpose of JSR? I cannot find either Hono nor ExpressJS on the website.

**Comment by BlackAsLight** [comment]
The packages available on NPM, JSR and Deno Land are the ones made available by their authors. In today's age you should look for packages on JSR first. If not there then fall back to NPM. Deno Land only has packages that were made with Deno in mind or made compatible with Deno, so you might find some older packages on Deno Land but not JSR or NPM.

The purpose of JSR is to be a modern package registry for JavaScript. It won't have a package if the author of said package doesn't choose to publish on JSR. JSR is still relatively new, and Deno are more focusing their attention around JSR as it allows them to deduplicate dependencies between versions.
